manager/manager_publish_test.go

```
@@ -0,0 +1,109 @@
+	m, err := New()
+	if err != nil {
+		t.Fatalf("new manager: %v", err)
+	}
+
+	// A cancelable context lets us deregister the subscriber on cleanup so the
+	// SSE handler goroutine can exit instead of leaking.
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	w := newStalledResponseWriter()
+	req := httptest.NewRequest(http.MethodGet, "/updates?stream=updates", nil).WithContext(ctx)
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)